eureka-core/src/main/java/com/netflix/eureka2/channel/AbstractClientChannel.java

```diff
@@ -23,25 +19,25 @@
 
     protected final TransportClient client;
 
-    /**
-     * There can only ever be one connection associated with a channel. This subject provides access to that connection
-     * after a call is made to {@link #connect()}
-     *
-     * Why is this a {@link ReplaySubject}?
-     *
-     * Since there is always every a single connection created by this channel, everyone needs to get the same
-     * connection. Now, the connection creation is lazy (in {@link #connect()} so we need a way to update this
-     * {@link Observable}. Hence a {@link Subject} and one that replays the single connection created.
-     */
-    private final ReplaySubject<MessageConnection> singleConnectionSubject;
+    private volatile MessageConnection connectionIfConnected;
 
-    private volatile MessageConnection connectionIfConnected; // External callers should use "singleConnectionSubject"
-    private final AtomicBoolean connectionRequestedOnce = new AtomicBoolean();
+    private final Observable<MessageConnection> singleConnection;
 
     protected AbstractClientChannel(final STATE initState, final TransportClient client) {
         super(initState);
         this.client = client;
-        singleConnectionSubject = ReplaySubject.create();
+
+        singleConnection = client.connect()
+                .take(1)
+                .map(new Func1<MessageConnection, MessageConnection>() {
+                    @Override
+                    public MessageConnection call(MessageConnection serverConnection) {
+                        connectionIfConnected = serverConnection;
+                        return connectionIfConnected;
+                    }
+                })
+                .replay()
```

Comment:
As this is ConnectableObservable, shouldn't we call connect on it? According to the documentation, this observable should not emit any items without this call.